File: lab3/s1/ejb/src/main/java/ru/uvuv643/ejb/remote/dto/human/request/DeleteHumanBeingRequest.java
package ru.uvuv643.ejb.remote.dto.human.request;

import ru.uvuv643.ejb.remote.dto.human.CarDto;

import jakarta.validation.Valid;
import  jakarta.xml.bind.annotation.XmlAccessType;
import  jakarta.xml.bind.annotation.XmlAccessorType;
import  jakarta.xml.bind.annotation.XmlElement;
import  jakarta.xml.bind.annotation.XmlRootElement;
import java.io.Serializable;

@XmlRootElement(name = "DeleteHumanBeingRequest")
@XmlAccessorType(XmlAccessType.FIELD)
public class DeleteHumanBeingRequest implements Serializable {

        @XmlElement(name = "limit")
        @Valid
        private Integer limit;

        @XmlElement(name = "car")
        @Valid
        private CarDto car;

        @XmlElement(name = "impactSpeed")
        private Float impactSpeed;

        public DeleteHumanBeingRequest() {
                this(null, null, null);
        }

        public DeleteHumanBeingRequest(Integer limit, CarDto car, Float impactSpeed) {
                this.limit = limit;
                this.car = car;
                this.impactSpeed = impactSpeed;
        }

        public Integer getLimit() {
                return limit;
        }

        public void setLimit(Integer limit) {
                this.limit = limit;
        }

        public CarDto getCar() {
                return car;
        }

        public void setCar(CarDto car) {
                this.car = car;
        }

        public Float getImpactSpeed() {
                return impactSpeed;
        }

        public void setImpactSpeed(Float impactSpeed) {
                this.impactSpeed = impactSpeed;
        }
}

File: lab2/s1/src/main/kotlin/ru/uvuv643/soa/api/v1/dto/human/request/DeleteHumanBeingRequest.kt
package ru.uvuv643.soa.api.v1.dto.human.request

import jakarta.annotation.Nullable
import jakarta.validation.Valid
import jakarta.validation.constraints.Pattern
import jakarta.xml.bind.annotation.XmlAccessType
import jakarta.xml.bind.annotation.XmlAccessorType
import jakarta.xml.bind.annotation.XmlElement
import jakarta.xml.bind.annotation.XmlRootElement
import ru.uvuv643.soa.api.v1.dto.human.CarDto

@XmlRootElement
@XmlAccessorType(XmlAccessType.FIELD)
data class DeleteHumanBeingRequest (

        @XmlElement(name = "limit")
        @field:Nullable
        @field:Valid
        val limit: Int?,

        @XmlElement(name = "car")
        @field:Nullable
        @field:Valid
        val car: CarDto?,

        @XmlElement(name = "impactSpeed")
        @field:Nullable
        val impactSpeed: Float?,

        ) {
        constructor() : this(null, null, null)
}